File: server/app/schemas/search_config.py
"""
Search configuration schemas.
"""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class SearchConfigBase(BaseModel):
    """Base schema for search configuration."""

    provider: str = "tavily"
    api_key: str = ""
    description: str = ""
    is_active: bool = False


class SearchConfigUpdate(BaseModel):
    """Schema for updating search configuration."""

    provider: Optional[str] = None
    api_key: Optional[str] = None
    description: Optional[str] = None
    is_active: Optional[bool] = None


class SearchConfigResponse(BaseModel):
    """Schema for search configuration response."""

    id: int
    provider: str
    api_key: str
    description: str
    is_active: bool
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True

File: server_go/internal/schema/search_config.go
package schema

import "time"

type SearchConfigUpdate struct {
	Provider    *string `json:"provider"`
	APIKey      *string `json:"api_key"`
	Description *string `json:"description"`
	IsActive    *bool   `json:"is_active"`
}

type SearchConfigResponse struct {
	ID          int64     `json:"id"`
	Provider    string    `json:"provider"`
	APIKey      string    `json:"api_key"`
	Description string    `json:"description"`
	IsActive    bool      `json:"is_active"`
	UpdatedAt   time.Time `json:"updated_at"`
}
